src/views/Header/Header.js

```diff
@@ -211,8 +211,18 @@ class Header extends React.PureComponent {
   };
 
   _renderRightComponent = props => {
-    const { headerRight } = props.scene.descriptor.options;
-    return headerRight || null;
+    const { options } = props.scene.descriptor;
+    const headerRight = options.headerRight;
+    if (React.isValidElement(headerRight) || headerRight === null) {
+      return headerRight;
+    }
+    if (!headerRight) return null;
+    return (
+      <headerRight
```

Comment:
This won't work, components need to be capitalized. Plus the headerRight is an element  (it is already rendered as JSX). and is not a component.

Reply:
I want to make headerRight similar to headerLeft.

As `headerLeft` option can receive both rendered JSX element and react component, so `headerRight` might also can do this.
If the provided headerRight is an element, there is no difference to previous code. Here is the code:
```
if (React.isValidElement(headerRight) || headerRight === null) {
    return headerRight;
}
```
If the provided headerRight is a React component, then we do the render work and renturn it.

Similar to `headerLeft`, both two ways of providing `headerRight` component below are supported:
```
static navigationOptions = {
    headerRight: <SomeHeaderRightButton />,
};
```
```
static navigationOptions = {
    headerRight: SomeHeaderRightButton,
};
```

The new way allows headerRight access `tintColor` prop passed down from upper.

I will add a line so that the rendered component become capitalized.

Hope you can understand me.